tests/flytekit/unit/core/test_array_node_map_task.py

```diff
@@ -464,3 +469,39 @@ def wf():
 
     with pytest.raises(AssertionError):
         wf.compile()
+
+
+def test_load_offloaded_literal(tmp_path, monkeypatch):
+    @task
+    def say_hello(name: str) -> str:
+        return f"hello {name}!"
+
+    ctx = context_manager.FlyteContextManager.current_context()
+    with context_manager.FlyteContextManager.with_context(
+            ctx.with_execution_state(
+                ctx.execution_state.with_params(mode=context_manager.ExecutionState.Mode.TASK_EXECUTION)
+            )
+    ) as ctx:
+        list_ints = ["a", "b", "c"]
+        lt = TypeEngine.to_literal_type(typing.List[str])
+        to_be_offloaded = TypeEngine.to_literal(ctx, list_ints, typing.List[str], lt)
+        with open(f"{tmp_path}/literal.pb", "wb") as f:
+            f.write(to_be_offloaded.to_flyte_idl().SerializeToString())
+
+        literal = Literal(
+            offloaded_metadata=LiteralOffloadedMetadata(
+                uri=f"{tmp_path}/literal.pb",
+                inferred_type=lt,
+            ),
+        )
+
+        lm = LiteralMap({
+            "name": literal
+        })
+
+        monkeypatch.setenv("BATCH_JOB_ARRAY_INDEX_VAR_NAME", "name")
+        monkeypatch.setenv("name", "0")
+        t = map_task(say_hello)
+        res = t.dispatch_execute(ctx, lm)
+        assert len(res.literals) == 1
+        assert res.literals["o0"].scalar.primitive.string_value == "hello a!"
```

Comment:
Can we verify that all 3 inputs work? This works by setting the `name` env var like this [test is doing](https://github.com/flyteorg/flytekit/pull/2771/files#diff-31caa03908ae2151c0e6160672d02a7e5e7c7a96e3c8a0a811020c744cf56bb2R503-R510).

Reply:
done